common/src/main/cpp/extensions/openxr_fb_scene_capture_extension_wrapper.cpp

```diff
@@ -96,16 +96,38 @@ void OpenXRFbSceneCaptureExtensionWrapper::_on_instance_destroyed() {
 	cleanup();
 }
 
-bool OpenXRFbSceneCaptureExtensionWrapper::request_scene_capture() {
-	XrAsyncRequestIdFB requestId;
+bool OpenXRFbSceneCaptureExtensionWrapper::request_scene_capture(String p_request, SceneCaptureCompleteCallback p_callback, void *p_userdata) {
+	if (scene_capture_enabled) {
+		ERR_PRINT("Already running scene capture");
+		p_callback(XR_ERROR_VALIDATION_FAILURE, p_userdata);
+		return false;
+	}
+
+	XrAsyncRequestIdFB request_id;
+
+	CharString cstr = p_request.ascii();
+
 	XrSceneCaptureRequestInfoFB request;
 	request.type = XR_TYPE_SCENE_CAPTURE_REQUEST_INFO_FB;
 	request.next = nullptr;
-	request.requestByteCount = 0;
-	request.request = nullptr;
-	XrResult result = xrRequestSceneCaptureFB((XrSession)get_openxr_api()->get_session(), &request, &requestId);
-	scene_capture_enabled = (result == XR_SUCCESS);
-	return scene_capture_enabled;
+	request.requestByteCount = cstr.size();
+	request.request = cstr.ptr();
+
+	XrResult result = xrRequestSceneCaptureFB((XrSession)get_openxr_api()->get_session(), &request, &request_id);
+	if (!XR_SUCCEEDED(result)) {
+		WARN_PRINT("xrRequestSceneCaptureFB failed!");
+		WARN_PRINT(get_openxr_api()->get_error_string(result));
+		p_callback(result, p_userdata);
+		return false;
+	}
+
+	scene_capture_enabled = true;
+	requests[request_id] = RequestInfo(p_callback, p_userdata);
```

Comment:
Seems like the only scenario where we can have multiple requests is when a scene capture fails and `scene_capture_enabled` is not set to `true`. Is that the intended behavior?

Reply:
OpenXR allows for multiple requests at a time, and so do a number of other similar asynchronous APIs, for which I've used this same pattern. So, I did it this way largely for consistency with all the other places we're doing asynchronous requests.

However, from a developer experience perspective, we don't want to allow multiple requests at a time - if the scene capture interface has opened up, we want to let that completely finish before making another request. And unless I made a mistake somewhere, the `scene_capture_enabled` flag should prevent multiple requests in all cases (even if we get an error - it'll either be reset or not yet set in the first place).

So, if we don't care about the consistency thing, we could drop the `HashMap` and just keep a single `RequestInfo *` for the current request? I could go either way. Which do you prefer?

Reply:
Let's keep it as is for consistency. 
Thanks for the clarification!